draft-ietf-httpbis-connect-tcp.md

```diff
@@ -62,41 +64,45 @@ This specification describes an alternative mechanism for proxying TCP in HTTP.
 
 # Specification
 
-A template-driven TCP transport proxy for HTTP is identified by a URI Template {{!RFC6570}} containing variables named "target_host" and "target_port".  This URI Template and its variable values MUST meet all the same requirements as for UDP proxying ({{!RFC9298, Section 2}}), and are subject to the same validation rules.  The client MUST substitute the destination host and port number into this template to produce the request URI.
+A template-driven TCP transport proxy for HTTP is identified by a URI Template {{!RFC6570}} containing variables named "target_host" and "target_port".  This URI Template and its variable values MUST meet all the same requirements as for UDP proxying ({{!RFC9298, Section 2}}), and are subject to the same validation rules.  The client MUST substitute the destination host and port number into this template to produce the request URI.  The derived URI serves as the destination of a Capsule Protocol connection using the Upgrade Token "connect-tcp" (see registration in {#new-upgrade-token}).
+
+When using "connect-tcp", TCP payload data is sent in the payload of a new Capsule Type named DATA (see registration in {{data-capsule}}).  The ordered concatenation of DATA capsule payloads represents the main payload data stream in any protocol where this is well-defined.  Intermediaries MAY split or merge DATA capsules.
```

Comment:
Two suggested changes here:
1. I think it would help to reuse the data stream concept from [the capsule document](https://www.rfc-editor.org/rfc/rfc9297#section-3.1)
2. I know this predates this PR, but the notion of "in any protocol where this is well-defined" is not fully clear to me. I think if we just cite the data stream then this becomes well defined anywhere where there is a data stream, which is anywhere there are no HTTP messages.

```suggestion
When using "connect-tcp", TCP payload data is sent in the payload of a new Capsule Type named DATA (see registration in {{data-capsule}}).  The ordered concatenation of DATA capsule payloads represents the data stream (as defined in {{Section 3.1 of !RFC9297}}).  Intermediaries MAY split or merge DATA capsules.
```

Reply:
I don't think we can say this.  All capsules live _inside_ the data stream, including.  DATA creates something like a new "nested" data stream whose meaning is specific to the Upgrade Token, but connect-tcp is the first Capsule Protocol Upgrade Token that has a stream of this kind.

I've rephrased this paragraph to avoid "in any protocol where this is well-defined" and tried to separate the speculative future applications from the specific use in this draft.